examples/gazebo_tank/gazebo_tank.cc

```diff
@@ -33,38 +34,48 @@ int bobMain(int argc, char **argv)
     /************************************Gazebo setup end************/
     std::unique_ptr<Video::Display> display;
     std::unique_ptr<Gazebo::Camera> cam;
-    if(argc >= 3) { // Initialize gazebo camera if more than 2 arguments are provided (display switch and camera url)
+    if (argc >= 3) { // Initialize gazebo camera if more than 2 arguments are provided (display switch and camera url)
         std::cout << "Display switch enabled.\n";
-        if(strcmp(argv[1], "-p") == 0) {
+        if (strcmp(argv[1], "-p") == 0) {
             std::cout << "Using panoramic camera.\n";
             cam = std::make_unique<Gazebo::Camera>(node, argv[2], true);
-            display = std::make_unique<Video::Display>(*cam, cv::Size(640,320)); //unwrap resolution needs to be supplied
-        }
-        else if(strcmp(argv[1], "-s") == 0){
+            display = std::make_unique<Video::Display>(*cam, cv::Size(640, 320)); //unwrap resolution needs to be supplied
+        } else if (strcmp(argv[1], "-s") == 0) {
             std::cout << "Using simple camera.\n";
             cam = std::make_unique<Gazebo::Camera>(node, argv[2], false);
             display = std::make_unique<Video::Display>(*cam); //unwrap resolution needs to be supplied
         }
     }
 
     Gazebo::Tank robot(1_mps, node); // Tank agent
-    HID::Joystick joystick(0.25f);
-    robot.controlWithThumbsticks(joystick);
-
-    std::cout << "Drive the car using the two thumbsticks: each stick is for one motor" << std::endl;
+    std::unique_ptr<HID::Joystick> joystick;
+    try {
+        joystick = std::make_unique<HID::Joystick>(0.25f);
+        robot.controlWithThumbsticks(*joystick);
+        std::cout << "Drive the car using the two thumbsticks: each stick is for one motor" << std::endl;
+    } catch (std::runtime_error &) {
+        LOGW << "Could not find joystick";
+    }
 
-    do {
-        // Check for joystick events
-        if (!joystick.update() && (!display || !display->update())) {
-            // A small delay so we don't hog CPU
-            std::this_thread::sleep_for(5ms);
-        }
-    } while (!joystick.isPressed(HID::JButton::B));
+    if (!display && !joystick) {
```

Comment:
What happens in either of these cases when you have a display but no joystick?

Reply:
If there's a display but no joystick, then you'll have an ``OpenCV`` window showing the current view and you can terminate the program by closing it.